### PR TITLE
Storybook: Remove flexbox from story container

### PR DESCRIPTION
**Pull request by joshhunt**
Removes flexbox from the parent elements of stories to fix some issues that make it difficult to debug and accurately represent components in Storybook.

We now have a stronger view that stories should represent their components as if they are placed on an empty page, defaulting to the top left corner of the document (unless something intrinsic to the component requires a difficult kind of layout, such as a Modal).

 - Change the styling in `withTheme` storybook decorator to just the bare minimum
 - Make `withCenteredStory` decorator a no-op. Later PR will remove this completely.
 - Remove `withPaddedStory` and `withRightAlignedStory` as they're not used anywhere
 - Fix issues in CodeEditor and InlineToast stories due to lack of flex layout

**Before:**
5943_2023-09-19-16-41_firefox

**After:**
5942_2023-09-19-16-41_firefox

### CODE REVIEW

File: packages/grafana-ui/src/components/InlineToast/InlineToast.internal.story.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ export const InlineToast: Story<InlineToastProps> = (args) => {
   const [el, setEl] = useState<null | HTMLInputElement>(null);
 
   return (
-    <div>
+    <div style={{ maxWidth: 500, width: `calc(100% - 100px)` }}>
```
Comment (L-M-K-B) [inline packages/grafana-ui/src/components/InlineToast/InlineToast.internal.story.tsx:36]: It's not `style={}` that makes the component look odd in storybook but something else.

In main branch: 
Screenshot 2023-09-20 at 09 03 23

In this branch:
Screenshot 2023-09-20 at 09 02 44

File: packages/grafana-ui/src/components/Monaco/CodeEditor.internal.story.tsx [hunk 2]
```diff
@@ -29,7 +28,7 @@ const meta: Meta<typeof CodeEditor> = {
 export const Basic: StoryFn<typeof CodeEditor> = (args) => {
   return (
     <CodeEditor
-      width={args.width}
+      width={'100%'}
```
Comment (L-M-K-B) [inline packages/grafana-ui/src/components/Monaco/CodeEditor.internal.story.tsx:31]:
```suggestion
      width='100%'
```

**Review by L-M-K-B — COMMENTED** [review]
That's a great start. Can you please have a look at the two things I noticed?

File: packages/grafana-ui/src/utils/storybook/withCenteredStory.tsx [hunk 3]
```diff
@@ -1,34 +1,13 @@
...
-        alignItems: vertical ? 'center' : 'flex-start',
-        justifyContent: horizontal ? 'center' : 'flex-start',
-      }}
-    >
-      {children}
-    </div>
-  );
-};
+/** @deprecated Stories should not be centered on the page. This decorator now does nothing */
+export const withNotCenteredStory: DecoratorFn = (story) => story();
```
Comment (tskarhed) [inline packages/grafana-ui/src/utils/storybook/withCenteredStory.tsx:4]: I assume we manually go through the stories to remove these from them in the future?

File: packages/grafana-ui/src/components/InlineToast/InlineToast.internal.story.tsx (hunk 1 above)
Comment (tskarhed) [reply packages/grafana-ui/src/components/InlineToast/InlineToast.internal.story.tsx:36]: This issue also appears for the AutoSaveField (which uses the InlineToast)

Comment (jackw) [reply packages/grafana-ui/src/components/InlineToast/InlineToast.internal.story.tsx:36]: I've run into this in the past with "pill" shaped elements. Pretty sure it's caused by the `border-radius: 100%` on the toast itself. Changing it from a percentage to a length like `100em` would probably solve it.

**Review by jackw — APPROVED** [review]
Code changes LGTM! Awesome work cleaning this up. 🚀

File: packages/grafana-ui/src/components/InlineToast/InlineToast.internal.story.tsx (hunk 1 above)
Comment (ashharrison90) [reply packages/grafana-ui/src/components/InlineToast/InlineToast.internal.story.tsx:36]: @JoaoSilvaGrafana fixed this yesterday: https://github.com/grafana/grafana/pull/75098#event-10411326589

it's just not on this branch

File: packages/grafana-ui/src/utils/storybook/withCenteredStory.tsx (hunk 3 above)
Comment (joshhunt) [reply packages/grafana-ui/src/utils/storybook/withCenteredStory.tsx:4]: yup!

**Review by tskarhed — APPROVED** [review]

**Pull request merged**